harsh.go

```diff
@@ -19,20 +19,37 @@ const (
 	ISO = "2006-01-02"
 )
 
+// Habit is where I started my review.  General  comment on naming right: Habit
+// vs. DailyHabit; whats relation?  Is daily habit a tracking of Habit?
+// Structs are not objects. I find the naming a bit confusing.  I then
+// immediate tried to find where and how these are used.
 type Habit struct {
 	name  string
 	every int
 }
 
+type Outcome string
+
+const (
+	// Success marks a completed habit based on your goals
+	Success Outcome = "y"
+
+	// Skipped means you didn't do it.  you may have had a good reason.  either way: it wasn't done
+	Skipped Outcome = "s"
+
+	// Warning is another outcome - this was hidden magic.
+	Warning = "w"
```

Comment:
@yarbelk Interesting. I hadn't thought of warning as an outcome actually. Outcome, in my mind, was the *explicit*  outcome the person entered about the habit by the person for the day. 

Warning was a courtesy sigil that happened depending on the condition of the outcome in relation to when it was last performed compared to how often someone said they wanted to perform it. (though, interestingly, yes they are generated the same way graph-wise. I guess I think of warning more as somethign the system does for you as a courtesy than as part of the data itself (w is never explicitly entered by someone into the system though that's an interesting idea.